### ISSUE TITLE
Ignore whitespace in (POI) search

### ISSUE THREAD

**Issue by sedrubal** [osmandapp/OsmAnd#13783]
### Description
Whitespaces should be ignored in the search. E.g. "burger heart" should also find "burgerheart" and "Goethe Straße" should also find "Goethestraße".

### How to reproduce?

Search for one of the terms mentioned before in Erlangen / Germany.

### Your Environment
OsmAnd Version: 4.1.11
Android/iOS version: Android 11 / LineageOS 18.1
Device model: One Plus 5

**Comment by Lee-Carre** [comment]
Sensible for several reasons.

Aside from typos and the like, also handles when OSM's `name=*` doesn't match user expectations.

Another example: PizzaExpress (a restaurant chain in Britain). To me, they're separate words, and I would habitually include a space between when searching.

Considering other issues (e.g. #12688) about name-matching, perhaps start by finding any matches without blank-space (the background isn't always white), to find exact matches early. Only then broaden the search to match against entries with spaces.

Mentioning Unicode Normalisation (there's a variety of spacing characters) is probably out-of-scope for this ticket, though may be relevant depending on implementation.

**Comment by vshcherb** [comment]
Related https://github.com/osmandapp/OsmAnd/issues/23709

**Comment by vshcherb** [comment]
Should be working already - "Goethe Straße" https://github.com/osmandapp/OsmAnd/issues/23709

**Comment by Zirochkabila** [comment]
# Testing 1

> OsmAnd~ 5.4.0#8575m, released: 2026-08-19
> OsmAnd Maps 5447, released 19.08.26

`49.5963, 11.0043`

| # | Search Query | Target Object | Expected Result | Status | Comment / Actual Result |
| :--- | :--- | :--- | :--- | :--- | :--- |
| **1a** | `Neustadter Kirchenplatz` | Square / Plaza | Finds target plaza | <span style="color:green; font-weight:bold;">PASSED 🟢</span> | Successfully locates `Neustädter Kirchenplatz`. |
| **1b** | `Neustadterkirchenplatz` | Square / Plaza | Ignores missing space | <span style="color:red; font-weight:bold;">FAILED 🔴</span> | Returns no results when space is omitted. |
| **2a** | `Franz Delitzsch` | Memorial / POI | Finds target memorial | <span style="color:green; font-weight:bold;">PASSED 🟢</span> | Successfully locates the memorial. |
| **2b** | `Franzdelitzsch` | Memorial / POI | Ignores missing space | <span style="color:red; font-weight:bold;">FAILED 🔴</span> | Returns no results when space is omitted. |




<details><summary>Details 1</summary>
<video src=""></video><table style="width:100%">
  <tr>
    <td><video source src="https://github.com/user-attachments/assets/a99da01a-fa2a-4997-a94d-bf39f785b218"  width="300" height="auto"/></td>
    <td><video source src="https://github.com/user-attachments/assets/dccc5eff-09ec-461d-be03-affff6742181"  width="300" height="auto"/> </td>
  </tr>
</table> 
</details> 


<details><summary>Details 2</summary>
<video src=""></video><table style="width:100%">
  <tr>
    <td><video source src="https://github.com/user-attachments/assets/3eedefdd-7993-4f05-8451-58260537127c"  width="300" height="auto"/></td>
    <td><video source src="https://github.com/user-attachments/assets/756ef756-d20b-4a6e-ae78-0d2510e3e8cd"  width="300" height="auto"/> </td>
  </tr>
</table> 
</details>

**Comment by vshcherb** [comment]
Web android needs to be tested - Spatial search

**Comment by Zirochkabila** [comment]
# Testing 2

> OsmAnd~ 5.4.0#8616m, released: 2026-08-25

Spatial search - **same result as in Testing 1**




<details><summary>Details</summary>
<table style="width:100%">
  <tr>
    <th>Neustadter Kirchenplatz and NeustadterKirchenplatz</th>
    <th>Franz Delitzsch and FranzDelitzsch</th>
  </tr>
  <tr>
    <td><video source src="https://github.com/user-attachments/assets/2c968e83-3ac9-4362-b6c7-03caea4be486"  width="300" height="auto"/></td>
    <td><video source src="https://github.com/user-attachments/assets/1b1e51d0-e7f9-456d-9690-1eeadd1a7267"  width="300" height="auto"/> </td>
  </tr>
</table> 
</details>

**Comment by vshcherb** [comment]
This is incorrect test @Zirochkabila "Goethe Straße"  - could be combined cause "Straße". 

Random word split is not supported, only meaningfull where 2nd part could be separate search word - https://github.com/osmandapp/OsmAnd/issues/25331 - see // TODO Short word split "Ro-ki" vs "Roki"

**Comment by Zirochkabila** [comment]
# Testing 3

> OsmAnd~ 5.4.0#8724m, released: 2026-09-08

**Location Context:** Wels, Upper Austria (`48.157, 14.028`)

| Search Query | Spatial Search | Result Proximity | Behavior / Root Cause | Status |
| :--- | :-: | :--- | :--- | :-: |
| `Goethestraße` | **OFF** | **Local (Wels)** | Exact string match with standard OSM compound name. | <span style="color:green; font-weight:bold;">PASSED 🟢</span> |
| `Goethe Straße` | **OFF** | **Far (>100 km)** | Tokenizer treats space as separator; falls back to global exact matches. | <span style="color:amber; font-weight:bold;">BY DESIGN 🟡</span> |
| `Goethe-Straße` | **OFF** | **Far (>100 km)** | Tokenizer treats hyphen as separator; falls back to global exact matches. | <span style="color:amber; font-weight:bold;">BY DESIGN 🟡</span> |
| `Goethestraße` | **ON** | **Local (Wels)** | Correctly prioritized by spatial index bounds. | <span style="color:green; font-weight:bold;">PASSED 🟢</span> |
| `Goethe Straße` | **ON** | **Local (Wels)** | Spatial proximity filter successfully matches separated tokens to local `Goethestraße`. | <span style="color:green; font-weight:bold;">PASSED 🟢</span> |
| `Goethe-Straße` | **ON** | **Local (Wels)** | Spatial proximity filter successfully matches hyphenated tokens to local `Goethestraße`. | <span style="color:green; font-weight:bold;">PASSED 🟢</span> |
